### ISSUE TITLE
[Question] Is Promise.all necessary on waiting for Events?

### ISSUE THREAD

**Issue by msmith1114** [microsoft/playwright#23967]
In the Documentation there is code that talks about clicking a button that triggers a request and then waiting on it (or any event). A sample would look something like this:

```
// Start waiting for request before clicking. Note no await.
const requestPromise = page.waitForRequest('https://example.com/resource');
await page.locator('button').click();
const request = await requestPromise;
```

Or Similarly maybe I see in a lot of places:

```
await page.locator('button').click();
await page.waitForRequest('https://example.com/resource');
```
(Sidenote are these basically identical as far as code goes?, do we get any benefit by setting the promise to a `const`)

However there are also a lot of guides that talk about using `Promise.all` to fix any flakyness with them. Such as https://tally-b.medium.com/the-illustrated-guide-to-using-promise-all-in-playwright-tests-af7a98af3f32 (Which does a good job of explaining potential problems)

I guess what I am curious of is this actually necessary? I believe Playwright mentioned using Promise.all in it's older documentation before but I don't see any references to it. Was something added behind the scenes to make `Promise.all` not really necessary?

As a side-question: If it's true......when would we ever want to use `Promise.all` in Playwright?

Thanks!

**Comment by mxschmitt** [comment]
We learned over the past few years that Promise.all is hard to understand for people who are new to JavaScript and new to async/await programming. (e.g. they put await inside the Promise.all body. Thats why we decided to drop Promise.all from our documentation, which has no logical changes but makes it easier for the user to understand on when a certain event handler starts and when it gets awaited.

Does it mean you don't need any Promise.all event handler? In most cases there is absolutely no need for Promise.all

Does this answer your question?

**Comment by pavelfeldman** [comment]
> Sidenote are these basically identical as far as code goes?, do we get any benefit by setting the promise to a const

These are not identical, in the second case w/o const, you can miss the request if it happens before click finishes.

**Comment by pavelfeldman** [comment]
> Or in the first example are we actually sending the request to the browser but just not waiting on the response until after the fact?

Pretty much, that's how JS works. `page.waitForRequest` installs request listeners. If you do that before click, you get the events, while if you do it after the click, you are registering your listener too late.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
